app/concepts/data_source/stock/tribunal_instance/unit/operation/load_transmission.rb

```diff
@@ -5,6 +5,36 @@ module Unit
         module Operation
           class LoadTransmission < Trailblazer::Operation
 
+            step Nested(ZIP::Operation::Extract)
+            fail :log_nested_error, fail_fast: true
+            step :check_file_count
+            fail :too_many_files, fail_fast: true
+            step ->(ctx, extracted_files:, **) { ctx[:extracted_file] = extracted_files.first }
+            pass :log_zip_info
+            step Nested(Import)
+            fail :log_nested_error
+
+            # TODO: find a better solution ?
+            step ->(ctx, dest_directory:, **) { FileUtils.rm_rf(dest_directory) }
+            fail ->(ctx, dest_directory:, **) { FileUtils.rm_rf(dest_directory) }
+
+            def check_file_count(ctx, extracted_files:, **)
+              extracted_files.count == 1
```

Comment:
C'est une contrainte métier l'unicité de fichier dans le zip ?

Reply:
non en effet c'était juste une constatation, je peux itérer sur tous les fichiers trouvés